### PR TITLE
feat(stdlib): Add `Exception.toString`

### PR DESCRIPTION
**Pull request by spotandjake**
This pr adds `Exception.toString` to the `Exception` module closing #1334 

Implementation notes, I decided to rework the way we handle exception printers, moving `panic` into its own module in `runtime/unsafe` and removing exception as a dependecy of `gc` and `malloc`. This allows us to use `Exception` in a regular mode and avoid a lot of the memory leaks we had before. This also allows us to address the `AssertionError` comment in the `Exception` file because we can allocate strings safely now. 

If we are not happy with this implementation I am happy to go back to how we handled things before and implement `Exception.toString` the way we did it before.

### CODE REVIEW

**Comment by ospencer** [comment]
I'm not against reworking the libraries in this way, but it borders on being a breaking change. It's only not technically breaking because there isn't a way to catch exceptions. I'd be more comfortable releasing this in a breaking release.

File: stdlib/runtime/unsafe/panic.gr [hunk 1]
```diff
@@ -0,0 +1,28 @@
+@runtimeMode
+module Panic
+
+from "runtime/unsafe/wasmi32" include WasmI32
+use WasmI32.{ (==), (+), (-) }
```
Comment (ospencer) [inline stdlib/runtime/unsafe/panic.gr:5]:
```suggestion
use WasmI32.{ (+) }
```

File: stdlib/exception.gr [hunk 2]
```diff
@@ -41,11 +41,16 @@ from "runtime/exception" include Exception
...
-  void
+  Exception.registerPrinter(printer)
 }
+
+/**
+ * Gets the string representation of the given exception.
+ *
+ * @param e: The exception to stringify
+ *
+ * @returns The string representation
```
Comment (ospencer) [inline stdlib/exception.gr:52]:
```suggestion
 * @returns The string representation of the exception
```

File: stdlib/exception.gr [hunk 3]
```diff
@@ -41,11 +41,16 @@ from "runtime/exception" include Exception
...
+ *
+ * @since v0.6.7
+ */
+provide let toString = e => Exception.toString(e)
```
Comment (ospencer) [inline stdlib/exception.gr:56]:
```suggestion
provide let toString = Exception.toString
```

File: stdlib/exception.gr [hunk 4]
```diff
@@ -41,11 +41,16 @@ from "runtime/exception" include Exception
  */
 @disableGC
```
Comment (ospencer) [inline stdlib/exception.gr:42]:
```suggestion
```
Honestly, this could just be a re-export like `toString` is.

File: stdlib/runtime/exception.gr [hunk 5]
```diff
@@ -1,96 +1,72 @@
...
-  WasmI32.store(iov, 1n, 12n)
-  fd_write(2n, iov, 2n, written)
-  unreachable()
+  exceptionToString(e, printers)
 }
 
+/**
+ * Throws an exception.
+ *
+ * @param e: The exception to throw
```
Comment (ospencer) [inline stdlib/runtime/exception.gr:62]:
```suggestion
 * Throws an uncatchable exception and traps.
 *
 * @param e: The exception to throw
```

File: stdlib/runtime/gc.gr [hunk 6]
```diff
@@ -21,21 +21,19 @@ module GC
...
 primitive (&&) = "@and"
 primitive (||) = "@or"
-primitive throw = "@throw"
 primitive ignore = "@ignore"
 primitive box = "@box"
 primitive unbox = "@unbox"
 
-exception DecRefError
+let throwDecRefError = () => {
+  Panic.panic("DecRefError: Attempted to decrement a zero reference count")
```
Comment (ospencer) [inline stdlib/runtime/gc.gr:35]:
```suggestion
  Panic.panic("DecRefError: Reference count of zero")
```

**Comment by spotandjake** [comment]
Made those changes

**Comment by ospencer** [comment]
Looks like you need to update the smallest program test.

**Comment by spotandjake** [comment]
Fixed

**Review by ospencer — APPROVED** [review]

**Pull request merged**